contracts/Orchestrator.sol

```diff
@@ -0,0 +1,165 @@
+     */
+    function addTransaction(address destination, uint value, bytes data)
+        external
+        onlyOwner
+    {
+        transactions.push(Transaction({
+            destination: destination,
+            value: value,
+            data: data,
+            enabled: true
+        }));
+    }
+
+    /**
+     * @param index Index of transaction to remove.
+     *              Transaction ordering may have changed since adding.
+     */
+    function removeTransaction(uint index)
+        external
+        onlyOwner
+    {
+        require(index < transactions.length, "index out of bounds");
+
+        if (index < transactions.length - 1) {
+            transactions[index] = transactions[transactions.length - 1];
+        }
+
+        delete transactions[transactions.length - 1];
+        transactions.length--;
+    }
+
+    /**
+     * @param index Index of transaction. Transaction ordering may have changed since adding.
+     * @param enabled True for enabled, false for disabled.
+     */
+    function setTransactionEnabled(uint index, bool enabled)
+        external
+        onlyOwner
+    {
+        require(index < transactions.length, "index must be in range of stored tx list");
+        transactions[index].enabled = enabled;
+    }
+
+    /**
+     * @return Number of transactions, both enabled and disabled, in transactions list.
+     */
+    function transactionsLength()
+        external
+        view
+        returns (uint256)
+    {
+        return transactions.length;
+    }
+
+    /**
+     * @dev wrapper to call the encoded transactions on downstream consumers.
+     * @param destination Address of destination contract.
+     * @param value ETH value
+     * @param dataLength Size of data param.
+     * @param data The encoded data payload.
+     * @return True on success
+     */
+    function externalCall(address destination, uint value, uint dataLength, bytes data)
+        internal
+        returns (bool)
+    {
+        bool result;
+        assembly {  // solhint-disable-line no-inline-assembly
+            // "Allocate" memory for output
+            // (0x40 is where "free memory" pointer is stored by convention)
+            let x := mload(0x40)
+
+            // First 32 bytes are the padded length of data, so exclude that
+            let d := add(data, 32)
+
+            result := call(
+                // TODO(iles): Update these gas numbers.
+
+
+
+                // 34710 is the value that solidity is currently emitting
+                // It includes callGas (700) + callVeryLow (3, to pay for SUB)
+                // + callValueTransferGas (9000) + callNewAccountGas
+                // (25000, in case the destination address does not exist and needs creating)
+                sub(gas, 34710),
+
+
+                destination,
+                value,
+                d,
+                dataLength,  // Size of the input (in bytes). This is what fixes the padding problem
+                x,
```

Comment:
How about emitting a log for successes as well, and make it include the the output

Reply:
would make rebase calls more expensive

Reply:
Yeah I'm really hesitant to add to the gas cost of rebase. If downstream wants to log, they can, but we're already opening to door to more gas with outside calls.